src/mui/input/SelectArrayInput.js

```diff
@@ -92,14 +88,12 @@ export class SelectArrayInput extends Component {
 
     handleAdd = newValue => {
         const values = [...this.state.values, newValue];
-        this.setState({ values });
-        this.handleChange(values);
+        this.setState({ values }, this.handleChange(values));
```

Comment:
if i remember correctly then second param to setState should be callback
ie. `this.setState({ values }, () => this.handleChange(values));`